src/payment.rs

```diff
@@ -1,4 +1,5 @@
 use cosmwasm_std::{Coin, MessageInfo, Uint128};
+use std::collections::HashSet;
```

Comment:
Did you check if it works in Wasm binary? I have reflection that because of hasher, it is impossible to use `HashMap` on wasm, and I am pretty sure the same thing would be with `HashSet`.

Reply:
Just tested. I compiled a contract that uses `HashSet<String>` with rust optimizer with no error. I also uploaded the wasm to osmosis testnet and instantiated the contract without issue as well.